src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -218,13 +218,117 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                         }
                         else
                         {
+#if UNIX
+                            // On UNIX systems, we expand arguments containing wildcard expressions against
+                            // the file system just like bash, etc.
```

Comment:
Are you sure? `/bin/echo *` should not echo files starting with `.`, and `/bin/echo "*"` should echo `*`, not the files in the current directory.

Maybe less interesting, but glob(7) says glob supports character classes like `[:upper:]` and more - we definitely don't support those. That said, it doesn't look like `bash` supports those.

Case-insensitive also concerns me. `rm M*` shouldn't remove files starting with lowercase `m`.

I'm not saying that glob(3) is exactly what we should use - just that there may be subtleties that we miss by not using a POSIX api.

As for PowerShell drives - I'm really curious if they'll be used on non-Windows platforms - I'm skeptical - symbolic links have served that purpose perfectly and work outside of PowerShell.